### PR TITLE
RSDK-1848: Don’t use video constraints for microphone

### PR DESCRIPTION
**Pull request by jckras**
Instead of using the default constraints (which were for video), microphone now uses audioConstraints. I was finally able to figure out how to test this by printing out the `malgo.DeviceInfo device.ID` and using that ID as the audio_path in app. 

-I tested the audio_input locally on darwin/arm64 (mac) by creating a microphone component and using the Macbook Air microphone as the audio stream. 

-I tested the audio_input locally on canon (for linux/arm64 and linux/amd64) by creating a microphone component and using the "Built-in Audio Analog Stereo Air" on the framework (aka framework microphone) microphone as the audio stream.

### CODE REVIEW

**Comment by jckras** [comment]
Audio_input running on macOS (I know you can't hear anything in a screenshot so let me know if you want video proof)
microphone!

**Comment by jckras** [comment]
Audio_input running on canon, specifically the framework here:
microphone framework

**Review by hexbabe — COMMENTED** [review]
This looks like it improves code correctness in the long run! I am lacking some context though. It seems that in the Jira comments, we are trying to avoid getting the `failed to find best driver to fit resource` error. That's what all the testing is for right, we are checking that a) the audio is playing b) the error is not there?

File: components/audioinput/microphone/microphone.go [hunk 1]
```diff
@@ -60,8 +60,17 @@ func newMicrophoneSource(conf *Config, logger logging.Logger) (audioinput.AudioS
 
 	debug := conf.Debug
 
+	var constraints mediadevices.MediaTrackConstraints
```
Comment (randhid) [inline components/audioinput/microphone/microphone.go:63]: Does this have a creator function? We could also initialise the empty variable `constraints := mediadevices.MediaTrackConstraints`

**Review by randhid — COMMENTED** [review]
Let's check with a microphone-enabled webcam on a raspberry pi, that's the mode that most surfaces the `cannot find driver that best fits constraints` error.

**Comment by jckras** [comment]
Audio_input running on a raspberry pis with a microphone-enabled webcam on a:
Screenshot 2024-10-09 at 5 18 40 PM
and there are no logs with `cannot find driver that best fits constraints`. Here are logs for reference:
Screenshot 2024-10-09 at 5 34 57 PM

**Review by seanavery — COMMENTED** [review]
Just getting up to speed with the code here myself 😅 

Perhaps not relevant to the PR here, but confused on why the attribute is named `audio_path` -- seems like we are matching with empty audio constraints and label/device-id.

**Review by randhid — APPROVED** [review]

File: components/audioinput/microphone/microphone.go (hunk 1 above)
Comment (seanavery) [reply components/audioinput/microphone/microphone.go:63]: Should be able to initialize this similar to the current (video) `DefaultConstraints`.

**Comment by jckras** [comment]
for reference I only changed files in microphone.go

**Pull request merged**